storage/remote/client.go

```diff
@@ -298,7 +309,11 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
-	defer cancel()
+	defer func() {
```

Comment:
I think it would be cleaner at this point to avoid copying cancel but explictly cancel on errors as needed as you did for status code

![image](https://github.com/prometheus/prometheus/assets/6950331/2cb93e49-52a6-457a-b081-613e47656570)

Reply:
Good point - will do!